source/windows/windows_pki_utils.c

```diff
@@ -603,7 +603,7 @@ int aws_import_key_pair_to_cert_context(
             aws_raise_error(AWS_ERROR_SYS_CALL_FAILURE);
         }
 
-        if (i != cert_count - 1 || !add_result) {
+        if (i != 0 || !add_result) {
```

Comment:
trivial: rename this PR something like "Fix certificate chains on windows"
because that's the real important change from the perspective of someone seeing this merge in the commit history.
The fact that you shuffled around some files for testing isn't as important from an outsider perspective, even if that's most of the changes in this PR